psl/psl.go

```diff
@@ -0,0 +1,171 @@
+// package psl defines the Psl struct containing alignment data as well as functions and methods that operate on the Psl struct
+package psl
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/fileio"
+	"strings"
+)
+
+// Psl struct declares data fields and types for Psl struct.
+type Psl struct {
+	Match       int
+	MisMatch    int
+	RepeatMatch int
+	Ns          int
+	QNumIns     int
+	QBaseIns    int
+	TNumIns     int
+	TBaseIns    int
+	Strand      string
+	QName       string
+	QSize       int
+	QStart      int
+	QEnd        int
+	TName       string
+	TSize       int
+	TStart      int
+	TEnd        int
+	BlockCount  int
+	BlockSize   []int
+	QList       []int
+	TList       []int
+}
+
+// PslReader is a struct that contains a SimpleReader with an embedded bufioReader and a 2-D byte slice to reduce memory allocation when reading each line.
+type PslReader struct {
+	Reader  *fileio.SimpleReader
+	curr    Psl
+	columns []string
+	err     bool
+}
+
+// NewPslReader will open a given text file and return a pointer to a PslReader struct.
+func NewPslReader(filename string) *PslReader {
+	var fields []string
+	var done bool
+	return &PslReader{
+		Reader:  fileio.NewSimpleReader(filename),
+		curr:    Psl{},
+		columns: fields,
+		err:     done,
+	}
+}
+
+// Read will process a given text file and parse the data fields of each line and return a slice of Psl structs.
+func Read(filename string) []Psl {
+	var ans []Psl
+	reader := NewPslReader(filename)
+	for curr, err := pslLine(reader); !err; curr, err = pslLine(reader) {
+		ans = append(ans, *curr)
+	}
+	return ans
+}
+
+// pslLine is a helper function which assigns data fields and return a pointer to a Psl.
```

Comment:
Would be good to also mention the bool that it returns, which I think is a flag for when it has completed reading the file.

Reply:
Yep I added that as a comment